### DISCUSSION TITLE
handling escalate action causes 'Maximum call stack size exceeded'

### DISCUSSION THREAD

**Discussion by kristofkalocsai** [statelyai/xstate#1550] [Q&A] [answered]
Not sure if I am using the escalate properly, but the general idea was:

1. child machines `escalate` errors

2. parent machine 'redirects' the escalated errors, namely that the `onError` handler of each child machine's node in the parent only transitions to an error state

3. and we handle these errors in one place.

However,
According to this exampe, https://codesandbox.io/s/staging-voice-91t0q?file=/src/index.js
This does not seem to be working.

Am I doing something wrong? Is this not possible to do?

Thanks

**Comment by davidkpiano** [comment, marked as answer]
The problem here has nothing to do with `escalate()`; rather, the `always` transition will always be taken in the `E` state (as the name implies).

Here's how to fix the infinite loop:

```diff
const machine = Machine({
  initial: "A",
  states: {
    A: {
      invoke: {
        src: childMachine,
        onError: {
          target: "E" // transitioning causes 'Maximum call stack size exceeded'
        }
      }
    },
    E: {
      always: {
        actions: (c, e) => console.log("E state", c, e),
+       target: "F"
      }
    },
+   F: {}
  }
  // ...
});
```

I'll add a warning for these scenarios.

**Reply by kristofkalocsai** [reply]
Oof, that was a big miss on my part.

Thanks a bunch!
